packages/date-time/src/components/Calendar/CalendarYear/CalendarYear.base.tsx

```diff
@@ -0,0 +1,458 @@
+        </div>
+      </FocusZone>
+    );
+  }
+  private _renderCell = (year: number): React.ReactNode => {
+    const { styles, theme, minYear, maxYear, onSelectYear, selectedYear } = this.props;
+    const selected = year === selectedYear;
+    const disabled = (minYear !== undefined && year < minYear) || (maxYear !== undefined && year > maxYear);
+    const current = year === new Date().getFullYear();
+    return (
+      <CalendarYearGridCell
+        {...this.props}
+        key={year}
+        year={year}
+        selected={selected}
+        current={current}
+        disabled={disabled}
+        onSelectYear={onSelectYear}
+        ref={selected ? this._onSelectedCellRef : current ? this._onCurrentCellRef : undefined}
+        styles={styles}
+        theme={theme}
+      />
+    );
+  };
+  private _onSelectedCellRef = (ref: CalendarYearGridCell) => {
+    this._selectedCellRef = ref;
+  };
+  private _onCurrentCellRef = (ref: CalendarYearGridCell) => {
+    this._currentCellRef = ref;
+  };
+}
+class CalendarYearNavPrev extends React.Component<ICalendarYearHeaderProps, {}> {
+  public render(): JSX.Element {
+    const { styles, theme, className, navigationIcons, strings, onSelectPrev } = this.props;
+
+    const classNames = getClassNames(styles, {
+      theme: theme!,
+      className: className
+    });
+
+    const iconStrings = navigationIcons || DefaultNavigationIcons;
+    const yearStrings = strings || DefaultCalendarYearStrings;
+    const prevRangeAriaLabel = yearStrings.prevRangeAriaLabel || yearStrings.rangeAriaLabel;
+    const prevAriaLabel = prevRangeAriaLabel
+      ? typeof prevRangeAriaLabel === 'string'
+        ? (prevRangeAriaLabel as string)
+        : (prevRangeAriaLabel as ICalendarYearRangeToString)(this.props)
+      : undefined;
+    const disabled = this.isDisabled;
+
+    return (
+      <button
+        className={css('ms-DatePicker-prevDecade', classNames.yearNavigationButton, {
+          [`ms-DatePicker-prevDecade--disabled ${classNames.disabledStyle}`]: disabled
+        })}
+        onClick={!disabled && onSelectPrev ? this._onSelectPrev : undefined}
+        onKeyDown={!disabled && onSelectPrev ? this._onKeyDown : undefined}
+        type="button"
+        tabIndex={0}
+        aria-label={prevAriaLabel}
+        disabled={disabled}
+      >
+        <Icon iconName={getRTL() ? iconStrings.rightNavigation : iconStrings.leftNavigation} />
+      </button>
+    );
+  }
+  get isDisabled(): boolean {
+    const { minYear } = this.props;
+    return minYear !== undefined && this.props.fromYear < minYear;
+  }
+  private _onSelectPrev = () => {
+    if (!this.isDisabled && this.props.onSelectPrev) {
+      this.props.onSelectPrev();
+    }
+  };
+  private _onKeyDown = (ev: React.KeyboardEvent<HTMLElement>) => {
+    if (ev.which === KeyCodes.enter) {
+      this._onSelectPrev();
+    }
+  };
+}
+class CalendarYearNavNext extends React.Component<ICalendarYearHeaderProps, {}> {
```

Comment:
Looks like we're missing some new lines between functions that we normally have.

Reply:
Direct copy from other component. Will fix.